python/sparkdl/transformers/named_image.py

```diff
@@ -161,6 +171,7 @@ def __init__(self, inputCol=None, outputCol=None, modelName=None, scaleHint="SCA
         """
         super(DeepImageFeaturizer, self).__init__()
         self._java_obj = self._new_java_obj("com.databricks.sparkdl.DeepImageFeaturizer", self.uid)
+        self._defaultParamMap = {} # discard unwanted defaults from parents
```

Comment:
There is a default from HasOutput which we do not want (the scala side has no default at least).

Reply:
this seems like a bug waiting to happen (person subclassing / taking a trait will expect to inherit the default values). do you think this needs to be overwritten generally, or just for the persistence (read) case? if the latter, we can handle this specific case in `_from_java` before `res._transfer_params_from_java()` is called.

(ugh it's weird / annoying that the scala version doesn't have a default value and the python version does)

Reply:
Actually I was wrong. Scala side has the same default. I got confused with the default params in the constructor. Good catch. I am moving it to the _from_java method.

Reply:
Ok, I got doubly confused. I was right the first time -  DeepImageFeaturizer  in scala does *NOT HAVE DEFAULT* for input/output column.

HasInputCol and HasOutputCol traits have default value. Python wrapper extends these traits , however scala DeepImageFeaturizer does not inherit either because these are not accessible in scala. The end result is incompatible defaults between scala and python.

I think the only reasonable fix is to not too extend HasInput/HasOutput in python so that python and scala are consistent.

Reply:
We actually copied over the python versions of `HasInputCol` and `HasOutputCol` from Spark to sparkdl, so we could do the same for Scala to ensure consistency.